lib/test.gi

```diff
@@ -209,6 +283,29 @@ end;
 ##  any directory in GAPInfo.KernelInfo.GAP_ROOT_PATHS is replaced with the
 ##  string GAPROOT/.
 ##  <P/>
+##  Lines which begin "#@" define special configuration options for tests.
+##  Currently defined options are:
+##  <List>
+##  <Mark>#@local identifierlist</Mark>
+##  <Item>Run the test as if it is in a function with local variable list
+##  <C>identifierlist</C>, which is a comma-separated list of
+##  identifers. Multiple #@local lines may be passed.
+##  </Item>
+##  <Mark>#@exec gapcode</Mark>
+##  <Item>Execute the code <C>gapcode</C> before the test is run. This allows
```

Comment:
```suggestion
##  <Item>Execute the code <C>gapcode</C> before any test in the input is run. This allows
```
Clarify that first *all* `#@exec` lines are run, before any actual test is performed. At least that is my understanding of what this PR does?

I wonder if that's really what we want, though? Why / how is this useful? You suggest that this allows "defining global variables when using #@local", but I don't understand why this is necessary? I see one concrete example of this in this PR:
```
#@local f
#@exec boolfalsevar := 0;
#@exec booltruevar := 0;
#@exec newtestvar := 0;
#@exec testvar := 0;
```
But why can't I just replace this by this?
```
#@local f
gap> boolfalsevar := 0;;
gap> booltruevar := 0;;
gap> newtestvar := 0;;
gap> testvar := 0;;
```

Reply:
That would produce `warning: Unbound global variable`. Of course, I could put in the definitions, and just put in what GAP would print (unbound global variable).

I was also planning (but never finished) running a test in parallel, where the execs would be run once and then the contents would be run in parallel on multiple threads (which currently almost always breaks, as tests define global variables and clash with each other).

Reply:
Oh, wait, so I now *must* use locals? There is no way to opt-in / opt-out? Wouldn't that then break the tests of all kinds of packages?

As to this specific use case: How about instead adding a `#@global boolfalsevar, booltruevar, ...` ?

Reply:
If you don't add local then everything works the same as before, so there is a difference between `#@local` (which means the test defines no variables) and not having a `#@local` statement, which keeps current behaviour.